Content.Client/Doors/Electronics/DoorElectronicsConfigurationMenu.xaml.cs

```diff
@@ -0,0 +1,46 @@
+using System.Linq;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Prototypes;
+using Content.Client.Access.UI;
+using Content.Client.Doors.Electronics;
+using Content.Shared.Access;
+using Content.Shared.Doors.Electronics;
+using FancyWindow = Content.Client.UserInterface.Controls.FancyWindow;
+
+namespace Content.Client.Doors.Electronics;
+
+[GenerateTypedNameReferences]
+public sealed partial class DoorElectronicsConfigurationMenu : FancyWindow
+{
+    private readonly DoorElectronicsBoundUserInterface _owner;
+    private AccessLevelControl _buttonsList;
+
+    public DoorElectronicsConfigurationMenu(DoorElectronicsBoundUserInterface ui, List<string> accessLevels, IPrototypeManager prototypeManager)
+    {
+        RobustXamlLoader.Load(this);
+
+        _owner = ui;
+
+        _buttonsList = new AccessLevelControl(accessLevels, prototypeManager);
+        AccessLevelControlContainer.AddChild(_buttonsList);
+
+        foreach (var (id, button) in _buttonsList.ButtonsList)
+        {
+            button.OnPressed += _ => UpdateConfiguration();
+        }
+    }
+
+    public void UpdateState(DoorElectronicsConfigurationState state)
+    {
+        _buttonsList.UpdateState(state.accessList);
+    }
+
+    private void UpdateConfiguration()
```

Comment:
this is used once so it isnt necessary to make a method, just use _owner.UpdateConfiguration above instead